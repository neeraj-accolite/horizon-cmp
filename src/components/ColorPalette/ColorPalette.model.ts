import { ReactNode } from 'react';

export interface ColorItemProps {
  title: string;
  colors: Record<string, string> | string;
}

export interface ColorPaletteBlockProps {
  name: string;
  color: string;
}

export interface ColorPaletteProps {
  children: ReactNode;
}

export interface TailwindThemeColorProps {
  extend: {
    colors: {
      [key: string]: Record<string, string> | string;
    };
  };
}
