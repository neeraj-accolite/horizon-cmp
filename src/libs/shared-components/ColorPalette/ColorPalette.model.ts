import { ReactNode } from 'react';

export interface ColorItemProps {
  title: string;
  colors: Record<string, string> | string;
}

export interface ColorPaletteBlockProps {
  name: string;
  color: string;
}

export interface ColorPaletteProps {
  children: ReactNode;
}
