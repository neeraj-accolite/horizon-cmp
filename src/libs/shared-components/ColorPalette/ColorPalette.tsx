import React from 'react';
import { ColorPaletteProps } from './ColorPalette.model';

const ColorPalette: React.FC<ColorPaletteProps> = ({ children }) => (
  <div>{children}</div>
);

export default ColorPalette;
