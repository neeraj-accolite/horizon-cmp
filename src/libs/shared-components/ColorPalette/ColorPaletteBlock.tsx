import React from 'react';
import { ColorPaletteBlockProps } from './ColorPalette.model';

const ColorPaletteBlock: React.FC<ColorPaletteBlockProps> = ({
  name,
  color,
}) => (
  <div key={name} className="mr-6 content-center justify-center text-center">
    <div className="flex content-center justify-center">
      <div
        className="size-12 border border-solid border-black"
        style={{
          backgroundColor: color,
        }}
      ></div>
    </div>
    <div>
      <span className="text-red-600">{name}</span>
    </div>
    <div>
      <span className="text-sm">{color}</span>
    </div>
  </div>
);

export default ColorPaletteBlock;
