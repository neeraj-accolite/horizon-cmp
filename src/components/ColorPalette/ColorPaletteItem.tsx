import React from 'react';
import { ColorItemProps } from './ColorPalette.model';
import ColorPaletteBlock from './ColorPaletteBlock';

const ColorPaletteItem: React.FC<ColorItemProps> = ({ title, colors }) => (
  <div className="mb-1">
    <div className="my-4 text-xl font-bold">{title}</div>
    <div className="flex content-center">
      {typeof colors === 'string' ? (
        <ColorPaletteBlock name={title} color={colors} />
      ) : (
        Object.entries(colors).map(([name, color]) => (
          <ColorPaletteBlock key={name} name={name} color={color} />
        ))
      )}
    </div>
  </div>
);

export default ColorPaletteItem;
