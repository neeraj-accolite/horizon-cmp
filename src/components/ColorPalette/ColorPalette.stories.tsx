import React from 'react';
import { Meta } from '@storybook/react';
import { theme as horizonTheme } from '../../../tailwind-configs/horizon-tailwind.config';
import { theme as casteelTheme } from '../../../tailwind-configs/casteel-tailwind.config';
import { theme as goldMedalTheme } from '../../../tailwind-configs/gold-medal-tailwind.config';
import { ColorItemProps, TailwindThemeColorProps } from './ColorPalette.model';
import ColorPalette from './ColorPalette';
import ColorPaletteItem from './ColorPaletteItem';

export default {
  title: 'Theme/ColorPalette',
  component: ColorPalette,
} as Meta;

const getThemeColors = (theme: TailwindThemeColorProps) => {
  const themeColors: ColorItemProps[] = [];
  Object.entries(theme.extend.colors).forEach(([key, value]) => {
    themeColors.push({
      title: key,
      colors: value,
    });
  });
  return themeColors;
};

const horizonThemeColors = getThemeColors(horizonTheme);

export const HorizonTheme = {
  render: () => (
    <ColorPalette>
      {horizonThemeColors.map((colorItem) => (
        <ColorPaletteItem
          key={colorItem.title}
          title={colorItem.title}
          colors={colorItem.colors}
        />
      ))}
    </ColorPalette>
  ),
};

const casteelThemeColors = getThemeColors(casteelTheme);
export const CasteelTheme = {
  render: () => (
    <ColorPalette>
      {casteelThemeColors.map((colorItem) => (
        <ColorPaletteItem
          key={colorItem.title}
          title={colorItem.title}
          colors={colorItem.colors}
        />
      ))}
    </ColorPalette>
  ),
};

const goldMedalThemeColors = getThemeColors(goldMedalTheme);
export const GoldMedalTheme = {
  render: () => (
    <ColorPalette>
      {goldMedalThemeColors.map((colorItem) => (
        <ColorPaletteItem
          key={colorItem.title}
          title={colorItem.title}
          colors={colorItem.colors}
        />
      ))}
    </ColorPalette>
  ),
};
