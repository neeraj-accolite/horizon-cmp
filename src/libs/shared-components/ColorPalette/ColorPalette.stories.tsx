import React from 'react';
import { Meta } from '@storybook/react';
import { ColorItemProps } from './ColorPalette.model';
import ColorPalette from './ColorPalette';
import ColorPaletteItem from './ColorPaletteItem';
import {
  MultiTenantThemeProps,
  horizonTheme,
  casteelTheme,
  goldMedalTheme,
} from '@theme-configs';

export default {
  title: 'Theme/ColorPalette',
  component: ColorPalette,
} as Meta;

const getThemeColors = (theme: MultiTenantThemeProps) => {
  const themeColors: ColorItemProps[] = [];
  Object.entries(theme.extend.colors).forEach(([key, value]) => {
    themeColors.push({
      title: key,
      colors: value,
    });
  });
  return themeColors;
};

const horizonThemeColors = getThemeColors(horizonTheme);

export const HorizonTheme = {
  render: () => (
    <ColorPalette>
      {horizonThemeColors.map((colorItem) => (
        <ColorPaletteItem
          key={colorItem.title}
          title={colorItem.title}
          colors={colorItem.colors}
        />
      ))}
    </ColorPalette>
  ),
};

const casteelThemeColors = getThemeColors(casteelTheme);
export const CasteelTheme = {
  render: () => (
    <ColorPalette>
      {casteelThemeColors.map((colorItem) => (
        <ColorPaletteItem
          key={colorItem.title}
          title={colorItem.title}
          colors={colorItem.colors}
        />
      ))}
    </ColorPalette>
  ),
};

const goldMedalThemeColors = getThemeColors(goldMedalTheme);
export const GoldMedalTheme = {
  render: () => (
    <ColorPalette>
      {goldMedalThemeColors.map((colorItem) => (
        <ColorPaletteItem
          key={colorItem.title}
          title={colorItem.title}
          colors={colorItem.colors}
        />
      ))}
    </ColorPalette>
  ),
};
