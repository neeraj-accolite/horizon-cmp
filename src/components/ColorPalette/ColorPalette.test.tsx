import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import ColorPalette from './ColorPalette';
import { ColorItemProps } from './ColorPalette.model';
import ColorPaletteItem from './ColorPaletteItem';

const themeColors: ColorItemProps[] = [
  {
    title: 'red',
    colors: '#FF0000',
  },
  {
    title: 'grey',
    colors: {
      'grey-100': '#f5f5f5',
      'grey-light': '#eeeeee',
    },
  },
];

const ColorPaletteComponent = () => {
  return (
    <ColorPalette>
      {themeColors.map((colorItem) => (
        <ColorPaletteItem
          key={colorItem.title}
          title={colorItem.title}
          colors={colorItem.colors}
        />
      ))}
    </ColorPalette>
  );
};

describe('Theme Component - Color Palette', () => {
  test('render the Color palette with theme of single color and multiple colors ', async () => {
    const container = render(<ColorPaletteComponent />);
    themeColors.map(async (colorItem) => {
      const colorPaletteTitleComponent = await container.findByText(
        colorItem.title,
      );
      expect(colorPaletteTitleComponent).toBeInTheDocument();
    });
  });

  test('The Color Palette snapshot is matching with the existing one ', async () => {
    const container = render(<ColorPaletteComponent />);
    expect(container).toMatchSnapshot();
  });
});
