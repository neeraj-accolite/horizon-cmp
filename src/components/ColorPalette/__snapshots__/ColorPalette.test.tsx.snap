// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme Component - Color Palette The Color Palette snapshot is matching with the existing one  1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="mb-1"
        >
          <div
            class="my-4 text-xl font-bold"
          >
            red
          </div>
          <div
            class="flex content-center"
          >
            <div
              class="mr-6 content-center justify-center text-center"
            >
              <div
                class="flex content-center justify-center"
              >
                <div
                  class="size-12 border border-solid border-black"
                  style="background-color: rgb(255, 0, 0);"
                />
              </div>
              <div>
                <span
                  class="text-medium"
                >
                  red
                </span>
              </div>
              <div>
                <span
                  class="text-sm"
                >
                  #FF0000
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="mb-1"
        >
          <div
            class="my-4 text-xl font-bold"
          >
            grey
          </div>
          <div
            class="flex content-center"
          >
            <div
              class="mr-6 content-center justify-center text-center"
            >
              <div
                class="flex content-center justify-center"
              >
                <div
                  class="size-12 border border-solid border-black"
                  style="background-color: rgb(245, 245, 245);"
                />
              </div>
              <div>
                <span
                  class="text-medium"
                >
                  grey-100
                </span>
              </div>
              <div>
                <span
                  class="text-sm"
                >
                  #f5f5f5
                </span>
              </div>
            </div>
            <div
              class="mr-6 content-center justify-center text-center"
            >
              <div
                class="flex content-center justify-center"
              >
                <div
                  class="size-12 border border-solid border-black"
                  style="background-color: rgb(238, 238, 238);"
                />
              </div>
              <div>
                <span
                  class="text-medium"
                >
                  grey-light
                </span>
              </div>
              <div>
                <span
                  class="text-sm"
                >
                  #eeeeee
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="mb-1"
      >
        <div
          class="my-4 text-xl font-bold"
        >
          red
        </div>
        <div
          class="flex content-center"
        >
          <div
            class="mr-6 content-center justify-center text-center"
          >
            <div
              class="flex content-center justify-center"
            >
              <div
                class="size-12 border border-solid border-black"
                style="background-color: rgb(255, 0, 0);"
              />
            </div>
            <div>
              <span
                class="text-medium"
              >
                red
              </span>
            </div>
            <div>
              <span
                class="text-sm"
              >
                #FF0000
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="mb-1"
      >
        <div
          class="my-4 text-xl font-bold"
        >
          grey
        </div>
        <div
          class="flex content-center"
        >
          <div
            class="mr-6 content-center justify-center text-center"
          >
            <div
              class="flex content-center justify-center"
            >
              <div
                class="size-12 border border-solid border-black"
                style="background-color: rgb(245, 245, 245);"
              />
            </div>
            <div>
              <span
                class="text-medium"
              >
                grey-100
              </span>
            </div>
            <div>
              <span
                class="text-sm"
              >
                #f5f5f5
              </span>
            </div>
          </div>
          <div
            class="mr-6 content-center justify-center text-center"
          >
            <div
              class="flex content-center justify-center"
            >
              <div
                class="size-12 border border-solid border-black"
                style="background-color: rgb(238, 238, 238);"
              />
            </div>
            <div>
              <span
                class="text-medium"
              >
                grey-light
              </span>
            </div>
            <div>
              <span
                class="text-sm"
              >
                #eeeeee
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
